devito/operator/profiling.py

```diff
@@ -420,13 +458,18 @@ def create_profile(name):
 
 def locate_intel_advisor():
     try:
-        path = Path(os.environ['ADVISOR_HOME'])
-        # Little hack: assuming a 64bit system
-        if path.joinpath('bin64').joinpath('advixe-cl').is_file():
-            return path
-        else:
-            warning("Requested `advisor` profiler, but couldn't locate executable")
-            return None
+        # Check if the directory to Intel Advisor is specified
+        path = Path(os.environ['DEVITO_ADVISOR_DIR'])
     except KeyError:
-        warning("Requested `advisor` profiler, but ADVISOR_HOME isn't set")
+        # Otherwise, automatically 'sniff' the location of Advisor's directory
+        path = sniff_advisor_location()
```

Comment:
can/should we merge the `sniff_...` and `locate_...` functions?

Reply:
Yes, they should be merged because there should be no situation where you use one and not the other and the latter is essentially almost a wrapper for the first. I will change that.